DontPanicLabs.Ifx.Telemetry.Logger.Azure.OpenTelemetry/Logger.cs

```diff
@@ -0,0 +1,149 @@
+﻿using Azure.Monitor.OpenTelemetry.Exporter;
+using DontPanicLabs.Ifx.Configuration.Local;
+using DontPanicLabs.Ifx.Telemetry.Logger.Azure.OpenTelemetry.Configuration;
+using DontPanicLabs.Ifx.Telemetry.Logger.Azure.OpenTelemetry.Exceptions;
+using DontPanicLabs.Ifx.Telemetry.Logger.Contracts;
+using Microsoft.Extensions.Logging;
+
+namespace DontPanicLabs.Ifx.Telemetry.Logger.Azure.OpenTelemetry;
+
+public class Logger : Contracts.ILogger
+{
+    private static ILoggerFactory? _LoggerFactory;
+    private static readonly object _LockObject = new();
+    private static bool _IsInitialized;
+
+    private readonly Microsoft.Extensions.Logging.ILogger _logger;
+
+    public Logger()
+    {
+        var openTelemetryConfig = new Config().GetOpenTelemetryConfiguration();
+
+        ConfigureTelemetry(openTelemetryConfig);
+
+        _logger = _LoggerFactory!.CreateLogger<Logger>();
+    }
+
+    // For testing purposes only
+    internal Logger(ILoggerFactory loggerFactory)
+    {
+        _logger = loggerFactory.CreateLogger<Logger>();
+    }
+
+    private static void ConfigureTelemetry(IOpenTelemetryConfiguration openTelemetryConfig)
+    {
+        if (_IsInitialized)
+        {
+            return;
+        }
+
+        lock (_LockObject)
+        {
+            if (_IsInitialized)
+            {
+                return;
+            }
+
+            EmptyConnectionStringException.ThrowIfEmpty(openTelemetryConfig.ConnectionString ?? "");
+
+            _LoggerFactory = LoggerFactory.Create(builder =>
+            {
+                builder.AddOpenTelemetry(options =>
+                {
+                    options.AddAzureMonitorLogExporter(exporterOptions =>
+                    {
+                        exporterOptions.ConnectionString = openTelemetryConfig.ConnectionString;
+                    });
+                    options.IncludeFormattedMessage = true;
+                    options.IncludeScopes = true;
+                });
+            });
+
+            _IsInitialized = true;
+        }
+    }
+
+    public void Log(string message, SeverityLevel severityLevel, IDictionary<string, string>? properties)
+    {
+        if (properties is { Count: > 0 })
+        {
+            throw new PlatformNotSupportedException("Trace Properties are not supported in OpenTelemetry.");
+        }
```

Comment:
Tarr / Tony and I have discussed this in the past, but to keep the same `ILogger` interface (which is app insights flavored), we'd have a couple options

1. Include the properties dictionary in the message by concat'ing everything (would be a bit hacky, and you wouldn't be able to do proper filtering in azure portal)
2. Throw platform not supported exception for operations that are not allowed (causes runtime errors and likely wouldn't appear until you hit QA)
3. Log the message, and then log a warning if properties are provided (semi-silent failure)

Or 

4. Don't use the same ifx `ILogger` interface (has many implications)

Reply:
Try 2, but create a new interface and mark the base method as [Obsolete]

Reply:
Ready for review